src/features/chart/redux/middleware.js

```diff
@@ -0,0 +1,14 @@
+import { UPDATE_CHART_RANGE } from "../../../store/constants";
+
+export const chartMiddleware = socketService => store => next => action => {
```

Comment:
FWIW:
I would change this method signature slightly:
```
export const chartMiddleware = ({ socketService }) => store => next => action => {
   ...
```
It'll allow you to extend that initial "dependencies" parameter, if you need to pass in additional services.

Reply:
Great point - will change